### PR TITLE
Extract file desc

### PR DESCRIPTION
**Pull request by anton-bondarev**

### CODE REVIEW

**Comment by Deryugin** [comment]
There are two `struct file`s left:
```
./src/fs/driver/ext2fuse/ext2fuse.c:17:static int ext2fuse_fill_sb(struct super_block *sb, struct file *bdev_file) {
./src/drivers/gpu/drm/etnaviv/etnaviv_dev.c:200:	struct file *file;
```

File: src/fs/dvfs/file_desc.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+off_t file_get_pos(struct file_desc *file) {
+	return file->pos;
+}
+
+off_t file_set_pos(struct file_desc *file, off_t off) {
+	file->pos = off;
+	return file->pos;
+}
+
+void *file_get_inode_data(struct file_desc *file) {
```
Comment (Deryugin) [inline src/fs/dvfs/file_desc.c]: assert(file);
assert(file->f_inode);

File: src/fs/dvfs/file_desc.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+	struct file_operations *f_ops;
+};
+
+extern off_t file_get_pos(struct file_desc *file);
+extern off_t file_set_pos(struct file_desc *file, off_t off);
+
+extern void *file_get_inode_data(struct file_desc *file);
+
+extern struct file_desc   *dvfs_alloc_file(void);
```
Comment (Deryugin) [inline src/fs/dvfs/file_desc.h]: `file_desc   *dvfs..`->`file_desc *dvfs...`

File: src/fs/dvfs/file_desc.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+extern int            dvfs_destroy_file(struct file_desc *desc);
```
Comment (Deryugin) [inline src/fs/dvfs/file_desc.h]: `int            dvfs..`->`int dvfs...`

Comment (anton-bondarev) [reply src/fs/dvfs/file_desc.h]: done

File: src/fs/dvfs/file_desc.h (hunk 2 above)
Comment (anton-bondarev) [reply src/fs/dvfs/file_desc.h]: done

File: src/fs/dvfs/file_desc.c (hunk 1 above)
Comment (anton-bondarev) [reply src/fs/dvfs/file_desc.c]: done

**Comment by anton-bondarev** [comment]
> There are two `struct file`s left:
> 
> ```
> ./src/fs/driver/ext2fuse/ext2fuse.c:17:static int ext2fuse_fill_sb(struct super_block *sb, struct file *bdev_file) {
> ./src/drivers/gpu/drm/etnaviv/etnaviv_dev.c:200:	struct file *file;
> ```

Thanks a lot

**Pull request merged**